main.cpp

```diff
@@ -1,31 +1,10 @@
 #include <iostream>
 #include <string>
-#include <curl/curl.h>
 #include "swApiClient.h"
 
-static size_t WriteCallback(void *contents, size_t size, size_t nmemb, void *userp)
-{
-    ((std::string*)userp)->append((char*)contents, size * nmemb);
-    return size * nmemb;
-}
-
 int main(void)
 {
-  CURL *curl;
-  CURLcode res;
-  std::string readBuffer;
-
-  swApiClient client("a", "b", "c");
-
-  curl = curl_easy_init();
-  if(curl) {
-    curl_easy_setopt(curl, CURLOPT_URL, "http://api.shapeways.com");
-    curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, WriteCallback);
-    curl_easy_setopt(curl, CURLOPT_WRITEDATA, &readBuffer);
-    res = curl_easy_perform(curl);
-    curl_easy_cleanup(curl);
-
-    std::cout << readBuffer << std::endl;
-  }
-  return 0;
+  swApiClient * client = new swApiClient("CLIENT_KEY", "CLIENT_SECRET");
```

Comment:
are those client_key and _secret the oauth2 variants of the Consumer Key and Consumer Secret, or the access token ? I think the Consumer Key and Secret. Let's use those names then.
Also lateron you use style 
swApiClient *client =

Reply:
it's consumer_key and consumer_secret: the .Authorize() function retrieves the bearer token.  I'll update that, and my styles :)